sdk/python/kubeflow/storage_init_container/hugging_face.py

```diff
@@ -57,3 +59,37 @@ def download_model_and_tokenizer(self):
         transformers.AutoTokenizer.from_pretrained(
             self.model, cache_dir=self.config.download_dir
         )
+
+
+@dataclass
+class HfDatasetParams:
+    repo_id: str
+    access_token: str = None
+    allow_patterns: list[str] = None
+    ignore_patterns: list[str] = None
+    download_dir: str = field(default="/workspace/datasets")
+
+    def __post_init__(self):
+        # Custom checks or validations can be added here
+        if self.repo_id is None:
```

Comment:
Should we verify that access token exists ?

Reply:
access_token is not needed for public datasets